### PR TITLE
Improve release steps ensuring stable official release branch

### PR DESCRIPTION
**Pull request by twelch**
Improve release instructions to ensure release branch is maintained.

- [x] Is this a bug fix, new functionality, or a breaking change?

Improve release instructions avoid release bugs

- [x] Have read and followed the steps for [preparing a pull request](https://github.com/Turfjs/turf/blob/master/CONTRIBUTING.md#preparing-a-pull-request).

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -212,14 +212,14 @@ Before release:
 
 Run the following release commands, replacing `7.0.0` with your version number:
 
-- create a release branch, replacing `xx` with your initials to make it clear who created the branch.
-  - `git checkout origin/master -b xx/release-7.0.0`
+- create a release branch
```
Comment (smallsaucepan) [inline CONTRIBUTING.md:215]: ... and switch to it. All building, tagging and publishing will be done on the release branch, and then merged back into master in a later step.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -228,16 +228,17 @@ Run the following release commands, replacing `7.0.0` with your version number:
   - Then reset by deleting your tag and branch locally and remotely.
     - `git push --delete origin v7.1.0`
     - `git tag --delete v7.1.0`
-    - `git push -d origin xx/release-7.0.0`
-    - `git branch -d xx/release-7.0.0`
+    - `git push -d origin release-7.0.0`
+    - `git branch -d release-7.0.0`
   - Now redo the steps above starting with creating "A clean local copy of the Turf Github repository"
 
 - If the release action was successful, now create a Pull Request for the release to update master.
```
Comment (smallsaucepan) [inline CONTRIBUTING.md:235]: .. now create a Pull Request to merge the release branch back into master.

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -228,16 +228,17 @@ Run the following release commands, replacing `7.0.0` with your version number:
...
   - You may be given a link in the output of the branch/tag push command to create the PR.
   - If you don't get this message, just go to https://github.com/Turfjs/turf/pulls and you should be prompted at the top to create a PR for this new branch you just pushed.
   - If that prompt doesn't appear, then just create a new pull request from the PR page and make sure the title is the version number e.g. `v7.0.0` and that it is merging your release branch -> to master.
   - Here is an example PR - https://github.com/Turfjs/turf/pull/2615.
-  - Get approval for the release PR and merge it!
+  - Get approval for the release PR, then "Squash and merge" it
+  - Click "Restore branch".  Github will have auto-deleted the release branch on merge.  Restoring is required because the release tag points at this release branch.
```
Comment (smallsaucepan) [inline CONTRIBUTING.md:241]: Is there anything we can do to stop this happening? I don't recall seeing this when I squash+merge other PRs.

**Review by smallsaucepan — APPROVED** [review]

**Comment by smallsaucepan** [comment]
Suggested some wording changes @twelch. If you think they're valuable feel free to incorporate.

Also, is there any way we can stop the release branch from being auto-deleted? Seems weirdly forceful of Github.

File: CONTRIBUTING.md (hunk 3 above)
Comment (twelch) [reply CONTRIBUTING.md:241]: My mistake, the auto-delete setting is not on for this repository (it's a top-level setting on the settings page).  I must have clicked to delete the branch after merge then.  I'll change the text accordingly.

**Comment by twelch** [comment]
Thanks for your feedback @smallsaucepan

**Pull request merged**